### DISCUSSION TITLE
Circuit?

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#29] [Q&A] [answered]
MIDI in should be connected to IO18 pin on the board right?
Anyone has a circuit diagram of this project?

**Comment by marcel-licence** [comment, marked as answer]
Yes when using the ESP32 Audio Kit
`#ifdef ESP32_AUDIO_KIT
#define MIDI_RX_PIN 18`
It will be IO18. All used IOs should be defined in config.h so you can map this as you like.

**Reply by jazzbiddy** [reply]
I can confirm IO18 on the ESP32 Audio Kit works with midi from an external board. I'm sending midi messages from an esp8266 and seeing all the messages logged by Marcel's code. Make sure your baud rates for the midi communication over serial match.

**Reply by souravlal** [reply]
Whats is the boad rate you are using for esp8266? Can you share your Code?

**Reply by jazzbiddy** [reply]
Here is a quick sample project you can run from another user (not my code). https://github.com/rjobidon/esp8266-midi-out. I connected a wire from D6 on the wemos to IO18 on the ESP Audio Kit, and wire from GND to both as well.   Then I uncommented the `//#define DUMP_SERIAL2_TO_SERIAL` line midi_interface.ino. after compiling and uploading, then powering both, I saw the midi notes logged from the Audio Kit. and I heard some sounds - which was more important to me! 

But I still don't have it working at all like Marcel's videos so for me there is more I need to learn and understand about the code.  When I'm stuck I try to search first on Google about the concepts and libraries that are used in this code so I can understand how they are used here.

**Reply by souravlal** [reply]
let me try

**Reply by marcel-licence** [reply]
Please check the line in config.h (I've updated it again because accidently I've checked in another setting)

`//#define MIDI_SERIAL2_BAUDRATE   115200 /* you can use this to change the serial speed */`

If this line is as shown then MIDI will default to the specified baud rate:

```
#ifndef MIDI_SERIAL2_BAUDRATE
#define MIDI_SERIAL2_BAUDRATE   31250
#endif
```
